test/e2e/network_segmentation_policy.go

```diff
@@ -271,6 +278,72 @@ var _ = ginkgo.Describe("Network Segmentation: Network Policies", func() {
 					return reachServerPodFromClient(cs, denyServerPodConfig, clientPodConfig, denyServerPodIP, port)
 				}, 1*time.Minute, 6*time.Second).ShouldNot(gomega.Succeed())
 
+				// Create client pod in red namespace and check network policy is working.
+				ginkgo.By("creating client pod in red namespace and check if it is in pending state until NAD is created")
+				clientPodConfig.namespace = namespaceRed
+				podSpec := generatePodSpec(clientPodConfig)
+				_, err = cs.CoreV1().Pods(clientPodConfig.namespace).Create(
+					context.Background(),
+					podSpec,
+					metav1.CreateOptions{},
+				)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				gomega.Consistently(func() v1.PodPhase {
+					updatedPod, err := cs.CoreV1().Pods(clientPodConfig.namespace).Get(context.Background(),
+						clientPodConfig.name, metav1.GetOptions{})
+					if err != nil {
+						return v1.PodFailed
+					}
+					return updatedPod.Status.Phase
+				}, 1*time.Minute, 6*time.Second).Should(gomega.Equal(v1.PodPending))
+
+				ginkgo.By("creating NAD for red and orange namespaces and check pod moves into running state")
```

Comment:
Can you add some comment here explaining what you are testing. Say something like "The pod won't run and the namespace address set won't be created until the NAD for the network is added to the namespace and we test here that once that happens the policy is reconciled to account for it "